### PR TITLE
Custom app URLs for ICDS

### PR DESCRIPTION
**Pull request by snopoke**
While looking into this it seemed easy enough to just do so I did.

@dimagi/scale-team

### CODE REVIEW

File: corehq/toggles.py [hunk 1]
```diff
@@ -724,3 +724,10 @@ def _commtrackify(domain_name, toggle_is_enabled):
     TAG_PRODUCT_PATH,
     [NAMESPACE_DOMAIN]
 )
+
+CUSTOM_APP_BASE_URL = StaticToggle(
+    'custom_app_base_url',
+    'Allow specifying a custom base URL for an application. Main use case is to allow migrating ICDS to a new cluster.',
```
Comment (dimagimon) [inline corehq/toggles.py]: E501 line too long (120 > 115 characters)

**Comment by czue** [comment]
nice, this looks good to me

File: corehq/apps/app_manager/models.py [hunk 2]
```diff
@@ -4074,13 +4075,13 @@ def form_record_url(self):
...
 
     @absolute_url_property
-    def hq_profile_url(self):
+    def profile_url(self):
         return "%s?latest=true" % (
             reverse('download_profile', args=[self.domain, self._id])
         )
 
     @absolute_url_property
-    def hq_media_profile_url(self):
```
Comment (dannyroberts) [inline corehq/apps/app_manager/models.py]: Any idea why this was separated out like this before?

**Comment by dannyroberts** [comment]
:star: Looks good to me. Had one sanity-check question, feel free to self-merge afterwards

**Comment by benrudolph** [comment]
:golf: nice!

File: corehq/apps/app_manager/models.py (hunk 2 above)
Comment (snopoke) [inline corehq/apps/app_manager/models.py]: I think your original intention was to support different URL's for remote apps but not 100% sure:
* https://github.com/dimagi/commcare-hq/commit/0a513fb5f23b1929e136597fcfb3545649c821fa

After that I think the pattern was just copied:
* https://github.com/dimagi/commcare-hq/commit/f77abe8a49d5dd55adf69667b6a50db0bb43e226

Comment (dannyroberts) [inline corehq/apps/app_manager/models.py]: Ha, ok that sounds right to me. Thanks!

**Pull request merged**